src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+use Zookeeper;
+use Throwable;
+
+/**
+ * ZookeeperStore is a StoreInterface implementation using Zookeeper as store engine.
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ */
+class ZookeeperStore implements StoreInterface
+{
+    /**
+     * @var \Zookeeper
+     */
+    private $zookeeper;
+
+    /**
+     * @var string A prefix for the locking resource to identify the process
+     */
+    private $prefix;
+
+    /**
+     * A base node within which all the locks for PHP application are managed in the Zookeeper environment.
+     */
+    private const LOCK_NODE = '/PHP_LOCK_NODE';
+
+    /**
+     * Locker constructor.
+     *
+     * @param \Zookeeper $zookeeper  Zookeeper Instance
+     * @param string     $lockPrefix Prefix for a locking resource
+     */
+    public function __construct(
+        Zookeeper $zookeeper,
+        string $lockPrefix = ''
+    ) {
+        $this->zookeeper = $zookeeper;
+        $this->prefix = $lockPrefix;
+    }
+
+    /**
+     * Stores the resource if it's not locked by someone else.
+     *
+     * @param \Symfony\Component\Lock\Key $key the Key that represent the lock resource
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public function save(Key $key)
+    {
+        $resource = $this->getKeyResource($key);
+
+        $lockNodePath = self::LOCK_NODE.'/'.$resource; // For example: /PHP_LOCK_NODE/MESSAGE_1234
+        // This makes sure the node dies at the end of a session, thus releasing the lock to avoid starvation
```

Comment:
The documentation says `node will automatically get removed if the client session goes away`. What is a session? What happens if the connection between the client and the server is broken? (For instance, the Zookeeper server restarts, a network issue, ....)